rfcs/0079-include-key-as.md

```diff
@@ -0,0 +1,89 @@
+---
+feature: New Syntax Include Key As
+authors:
+  - "TabVersion"
+start_date: "2023/11/22"
+---
+
+# New Syntax: Include \<column\> as \<new name\>
+
+## Background
+
+> This RFC is a follow-up to [RFC-0064](./0064-new-source-ddl.md).
+
+In previous implementation, we only load message key into a [~~hidden~~](https://github.com/risingwavelabs/risingwave/pull/13521) 
+column called `_rw_key` as `bytea` on default.
+
+Here are some cases when RisingWave derives the `_rw_key` column.
+
+* For all connectors with `format upsert`, RisingWave derives the column as primary key to perform upsert semantic.
+  * An exception: for `format upsert encode json`, RisingWave allows the PK to be part of the message payload instead of
+  the message key. This behavior is more like `format plain encode json` with PK constraint.
+  * **To Be Discussed**: whether we can afford to perform a breaking change for the above exception.
```

Comment:
> * An exception: for `format upsert encode json`, RisingWave allows the PK to be part of the message payload instead of the message key. This behavior is more like `format plain encode json` with PK constraint.

IIRC, according to our previous discussion, this will be `format insert` + normal PK definition i.e. `primary key (foo, bar)`. 

The exception looks very inconsistent and I'd like to get rid of it.

Reply:
In some cases, user want to determine the primary key of the table for some use cases such as 
- better batching/ serving performance
- streaming temporal join
And in those cases, user can make sure the primary keys column is exactly the message key in Kafka.

Reply:
If `format upsert` is specific to MQ with delete tombsonte (null value), I think the PK must be the MQ message key and cannot be a field in the value.

Reply:
I think either of the following behavior is acceptable:

1. All fields in message key are used as PK, and empty message body will be considered as delete tomb 
2. Some fields in message body are used as PK, and empty message body will not be considered as delete tomb.

1 corresponds to `upsert` semantics, while 2 corresponds to `plain`/`insert` semantics